pkg/resources/common/formatter.go

```diff
@@ -184,3 +222,24 @@ func excludeValues(request *types.APIRequest, unstr *unstructured.Unstructured)
 		}
 	}
 }
+
+func getNamespaceFromResource(obj types.APIObject) string {
+	unstr, ok := obj.Object.(*unstructured.Unstructured)
+	if !ok {
+		return ""
+	}
+
+	// If we have a backingNamespace, use that
+	if statusRaw, ok := unstr.Object["status"]; ok {
+		if statusMap, ok := statusRaw.(map[string]interface{}); ok {
+			if backingNamespace, ok := statusMap["backingNamespace"].(string); ok && backingNamespace != "" {
+				return backingNamespace
+			}
+		}
+	}
+
+	// Otherwise, if the id has a slash, we will interpret that.
+	// This is used to determine a project's namespace when there is no backingNamespace present.
+	// For cases where there is no slash, we use the object's ID, which is the same as the namespace.
+	return strings.Replace(obj.ID, "/", "-", 1)
```

Comment:
I hit this URl https://localhost:8443/v1/apps.deployments?checkPermissions=namespaces

and I get the following result 

<img width="719" alt="Screenshot 2025-05-06 at 11 48 52 AM" src="https://github.com/user-attachments/assets/7fc90df5-ba4d-4eaa-9217-2312cb3b37a4" />


There is no such namespace and the link also doesn't work. 

My URL is logically wrong, but when I put something logically wrong, it should not show these wrong links or `resourcePermissions` shouldn't be shown. Let me know your thoughts on this.

Reply:
Interesting.  I did leave it intentionally open to being able to request things that might not make sense, but that is a surprising result.  I'll take a look to see if I can make it so it doesn't return those obviously incorrect resourcePermissions.  Definitely open to ideas if you have any.

Reply:
Ah...I bet you were logged-in as admin.  For admin, the answer to any permission request is always, "yes".

Reply:
full circle here..../v1/apps.deployments will give a list of all deployments......then the ensuing checks will all return true for admin.  Definitely not the case for normal users.